### PR TITLE
[#1184] Admin Donations Search

### LINKED ISSUE #1184: [Feature] Filter&Search for Donations Grid in Admin View
**Is your feature request related to a problem? Please describe.**
With the growing count of campaigns and donations the need comes for having better ways of searching and filtering of the donation grid in Admin View. Currently the grid shows all donations X per page, but they can only be sorted and not filtered or searched.

**Describe the solution you'd like**
When opening /admin/donations the page to allow:

- [ ] default page size to be 20 and selectable by dropdown
- [ ]  filtering of donations by status(deleted hidden by default)
- [ ]  filtering of donations by type
- [ ]  filtering of donations by date created(range)
- [ ]  Search by names, email

**Describe alternatives you've considered**
The filters could be on the grid column titles or as separate toolbar above the grid

**Additional context**

### PR DESCRIPTION
**Pull request by AndreyGoranov**
Closes #1184

This PR is part of #1184 and it contains the logic for Searching through the Admin Donations Grid
This PR relies on https://github.com/podkrepi-bg/api/pull/413
## Screenshots:
image

### New environment variables:

- [ ] `NEW_ENV_VAR`: env var details

### New or updated dependencies:

Dependency name|Previous version|Updated version|Details
---|---|---|---
`dependency/name`|`v1.0.0`|`v2.0.0`|

### CODE REVIEW

File: src/components/donations/grid/GridAppbar.tsx [hunk 1]
```diff
@@ -28,6 +32,18 @@ export default function GridAppbar() {
       AlertStore.show(t('common:alerts.success'), 'success')
     },
   })
+  const [searchValue, setSearchValue] = useState('')
+
+  const debounceSearch = debounce((text: string) => {
+    donationStore.setDonationSearch(text)
+  }, 1500)
```
Comment (slavcho) [inline src/components/donations/grid/GridAppbar.tsx:39]: I would make this 0.5 or 1 sec.

**Review by slavcho — APPROVED** [review]
Looks good to me.

File: src/components/donations/grid/GridAppbar.tsx (hunk 1 above)
Comment (AndreyGoranov) [reply src/components/donations/grid/GridAppbar.tsx:39]: Yea, actually I don't know why I left it at 1500ms. 500ms will be more than enough.

**Review by igoychev — APPROVED** [review]
you can't imagine how much this helps to track donations on Admin side!
@Juicebgs will be very happy!

**Pull request merged**